services/movie_session.py

```diff
@@ -0,0 +1,48 @@
+from django.db.models.query import QuerySet
+
+from db.models import MovieSession, CinemaHall, Movie
+from datetime import datetime
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall=CinemaHall.objects.get(id=cinema_hall_id),
+        movie=Movie.objects.get(id=movie_id)
+    )
+
+
+def get_movies_sessions(session_date: datetime = None) -> QuerySet:
+    if session_date:
+        return MovieSession.objects.filter(show_time__date=session_date)
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None
+) -> None:
+    session = MovieSession.objects.filter(id=session_id)
+    if show_time:
+        session.update(show_time=show_time)
+
+    if movie_id:
+        session.update(movie=movie_id)
+
+    if cinema_hall_id:
+        session.update(cinema_hall=cinema_hall_id)

```

Comment:
I think you need to use `session.save()`

Reply:
Not in this case - I used ORM manager (objects) for update - it added changes itself.
You need to use save() when trying to add changes through the class instance:

session = MovieSession.objects.get(id=session_id) - become instance
session.show_time = show_time - add changes
session.save( )

Reply:
Not in this case - I used ORM manager (objects) for update - it added changes itself.
You need to use save() when trying to add changes through the class instance:

session = MovieSession.objects.get(id=session_id) - become instance
session.show_time = show_time - add changes
session.save( )